.github/workflows/hail-search-unit-tests.yaml

```diff
@@ -0,0 +1,30 @@
+name: Hail Search Unit Tests
+
+# Run the test suite on pushes (incl. merges) to master and dev
+# Run the test suite when a PR is opened, pushed to, or reopened
+on:
+  push:
+    branches:
+      - master
+      - dev
+    paths:
+      - 'hail_search/**'
```

Comment:
Split the `hail_search` unit tests out into their own GitHub action so that we can trigger the docker build based on this `paths_switch`.

Reply:
If we update the docker image but not the search code it looks like this won't get run. Could we also add the path to the dockerfile here?

Reply:
yeah, we can add no problem, but do we want to add something in here that actually uses/builds the docker image?

Reply:
I guess not, I don't think we actually need to run the unit tests if the docker image changes, but I guess I was trying to solve the issue that the docker build is only triggered based on this job. If theres a different way to trigger building a docker image when the dockerfile is updated, that seems better to me

Reply:
Maybe splitting out the `Dockerfile Linting` job into one-per service also makes sense?

Reply:
actually, what about just moving the `Dockerfile` to a `deploy` directory inside `hail_search`?

Reply:
yes! I like this